examples/checkpointing/checkpoint.py

```diff
@@ -0,0 +1,65 @@
+from pyrevolve import Checkpoint, Operator
+from devito import TimeFunction
+
+
+class DevitoOperator(Operator):
+    """Class to wrap devito operators so they conform to the pyRevolve API
+    """
+    def __init__(self, op, args, argnames):
+        self.op = op
+        self.args = args
+        self.argnames = argnames
+
+    def apply(self, t_start, t_end):
+        """ If the devito operator requires some extra arguments in the call to apply
+            they can be stored in the args property of this object so pyRevolve calls
+            pyRevolve.Operator.apply() without caring about these extra arguments while
+            this method passes them on correctly to devito.Operator
+        """
+        args = self.args.copy()
+        args[self.argnames['t_start']] = t_start
+        args[self.argnames['t_end']] = t_end
+        self.op.apply(**args)
+
+
+class DevitoCheckpoint(Checkpoint):
```

Comment:
so, again, why don't we put all this stuff in a ``checkpointing.py`` module, rather than as an example?

Reply:
This is a `checkpointing.py` module, it is just called `checkpoint.py` (if it is the name you are concerned about, I am happy to change). Checkpointing is not a feature that devito provides. Pyrevolve provides the feature to implement checkpointing with just a few lines, given you have the appropriate operators (that devito provides). This is an example of using devito and pyrevolve together. Just like `examples/seismic/operators.py` etc., `examples/checkpointing/` is an appropriate place for it, no?

Reply:
I wasn't concerned about the name. OK, I have a different view about this, but I may well be wrong:

> Checkpointing is not a feature that devito provides

I thought checkpointing is a feature that Devito **provides**; it provides it through ``pyrevolve``, but that's a different story. This way, any Devito users would be able to use checkpointing, as long as a checkpointable Operator were used. Or, perhaps, all Operators are checkpointable... anyway, let me think about this

Reply:
What are you suggesting here anyway?